circuit_knitting/cutting/cutting_evaluation.py

```diff
@@ -306,13 +367,21 @@ def _generate_cutting_experiments(
             decomp_qc = decompose_qpd_instructions(
                 subcircuit, subcirc_qpd_gate_ids[i], map_ids_tmp
             )
-            subexperiments[-1].append([])
+            subexperiments_legacy[-1].append([])
             so = subsystem_observables[label]
             for j, cog in enumerate(so.groups):
                 meas_qc = _append_measurement_circuit(decomp_qc, cog)
-                subexperiments[-1][-1].append(meas_qc)
+                subexperiments_legacy[-1][-1].append(meas_qc)
 
-    return subexperiments, coefficients, sampled_frequencies
+    # If the circuit wasn't separated, return the subexperiments as a list
+    subexperiments_out: list[QuantumCircuit] | dict[
+        str | int, list[QuantumCircuit]
+    ] = subexperiments_dict
+    assert isinstance(subexperiments_out, dict)
+    if len(subexperiments_out.keys()) == 1:
+        subexperiments_out = subexperiments_dict[list(subexperiments_dict.keys())[0]]
```

Comment:
I think it is better to look at the input type of `circuits` here rather than the length of unique labels.  If somebody gave us a `dict`, we should give them a `dict`, too, in return, even if their `dict` contained only one element.  Or, in other words, the return type should depend on the _type_ of the input, not the _value_ of the input (a principle known as "type stability").  We should add a test for this case, too.

```suggestion
    if isinstance(circuits, QuantumCircuit):
        assert len(subexperiments_out.keys()) == 1
        subexperiments_out = list(subexperiments_dict.values())[0]
```